pkgdb/include/flox/lock-flake-installable.hh

```diff
@@ -0,0 +1,134 @@
+/* ========================================================================== *
+ *
+ * @file flox/lock-flake-installable.hh
+ *
+ * @brief Executable command helper and `flox::lockFlakeInstallable`.
+ *
+ *
+ * -------------------------------------------------------------------------- */
+
+#pragma once
+
+#include <filesystem>
+
+#include <argparse/argparse.hpp>
+
+#include "flox/core/command.hh"
+#include "flox/core/nix-state.hh"
+
+
+/* -------------------------------------------------------------------------- */
+
+namespace flox {
+
+/* -------------------------------------------------------------------------- */
+
+/** @brief Lock a flake installable for flox */
+class LockCommand : flox::NixState
+{
+
+private:
+
+  command::VerboseParser parser;
+
+  std::string installable;
+  std::string system = nix::settings.thisSystem.get();
+
+public:
+
+  LockCommand();
+
+  [[nodiscard]] command::VerboseParser &
+  getParser()
+  {
+    return this->parser;
+  }
+
+  /**
+   * @brief Execute the `lock` routine.
+   * @return `EXIT_SUCCESS` or `EXIT_FAILURE`.
+   */
+  int
+  run();
+};
+
+
+/* -------------------------------------------------------------------------- */
+
+struct LockedInstallable
+{
+  std::string                        lockedUrl;
+  std::optional<std::string>         flakeDescription;
+  std::string                        lockedAttrPath;
```

Comment:
nonblocking: we've discussed and settled on the terminology we should use to differentiate between partial and full attr path is `attr_path` and `flake_attr_path`
```suggestion
  std::string                        lockedFlakeAttrPath;
```